test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/RazorViewEngineOptionsSetupTest.cs

```diff
@@ -28,5 +33,132 @@ public void RazorViewEngineOptionsSetup_SetsUpFileProvider()
             var fileProvider = Assert.Single(options.FileProviders);
             Assert.Same(expected, fileProvider);
         }
+
+        [Fact]
+        public void PostConfigure_SetsAllowRecompilingViewsOnFileChange_For21()
+        {
+            // Arrange
+            var options = new RazorViewEngineOptions();
+            var compatibilityVersion = new MvcCompatibilityOptions { CompatibilityVersion = CompatibilityVersion.Version_2_1 };
+            var optionsSetup = GetSetup(compatibilityOptions: compatibilityVersion);
```

Comment:
Suggest simplifying the boilerplate and pass the `CompatibilityVersion` to `GetSetup(...)`.
(Also an odd name for a `MvcCompatibilityOptions` instance.)